pkg/controller/restore-operator/http.go

```diff
@@ -0,0 +1,75 @@
+// Copyright 2017 The etcd-operator Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package controller
+
+import (
+	"fmt"
+	"net/http"
+
+	api "github.com/coreos/etcd-operator/pkg/apis/etcd/v1beta2"
+	"github.com/coreos/etcd-operator/pkg/backup"
+	"github.com/coreos/etcd-operator/pkg/backup/backupapi"
+	"github.com/coreos/etcd-operator/pkg/controller/controllerutil"
+
+	"github.com/sirupsen/logrus"
+)
+
+const (
+	backupHTTPPath = "/backup"
+	listenAddr     = "0.0.0.0:19999"
+)
+
+func (r *Restore) startHTTP() {
+	http.HandleFunc(backupapi.APIV1+"/backup", r.serveBackup)
+	logrus.Infof("listening on %v", listenAddr)
+	panic(http.ListenAndServe(listenAddr, nil))
+}
+
+// serveBackup parses incoming request url of the form /backup/<cluster-name>
+// get the etcd cluster name.
+// Then it returns the etcd cluster backup snapshot to the caller.
+func (r *Restore) serveBackup(w http.ResponseWriter, req *http.Request) {
+	clusterName := string(req.URL.Path[len(backupHTTPPath+"/")])
+	if len(clusterName) == 0 {
+		http.Error(w, "cluster is not specified", http.StatusBadRequest)
+		return
+	}
+	v, ok := r.restoreCRs.Load(clusterName)
+	if !ok {
+		http.Error(w, fmt.Sprintf("cluster %v backup server not found", clusterName), http.StatusInternalServerError)
+		return
+	}
+	// a new go routine is created to handle each request to speed up
+	// multi requests handling.
+	go func() {
```

Comment:
Again, don't create a goroutine here.
The performance concern is unnecessary for now.

Reply:
done.